File: push/src/main/java/com/xiaomi/push/service/profile/MessageProfiling.java
package com.xiaomi.push.service.profile;

import android.util.Pair;
import com.xiaomi.push.mpcd.Constants;
import java.util.Vector;
import java.util.concurrent.ConcurrentHashMap;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/service/profile/MessageProfiling.class */
public class MessageProfiling {
    private static Vector<Pair<String, Long>> sSentPerfDatas = new Vector<>();
    private static ConcurrentHashMap<String, Long> sSendingMessages = new ConcurrentHashMap<>();

    public static String getPrefString() {
        StringBuilder sb = new StringBuilder();
        synchronized (sSentPerfDatas) {
            for (int i = 0; i < sSentPerfDatas.size(); i++) {
                Pair<String, Long> pairElementAt = sSentPerfDatas.elementAt(i);
                sb.append((String) pairElementAt.first);
                sb.append(":");
                sb.append(pairElementAt.second);
                if (i < sSentPerfDatas.size() - 1) {
                    sb.append(Constants.ITEM_SEPARATOR);
                }
            }
            sSentPerfDatas.clear();
        }
        return sb.toString();
    }

    public static void onReceiveSentAck(String str, String str2) {
        if (sSendingMessages.containsKey(str)) {
            long jLongValue = sSendingMessages.get(str).longValue();
            sSentPerfDatas.add(new Pair<>(str2, Long.valueOf(System.currentTimeMillis() - jLongValue)));
            sSendingMessages.remove(str);
        }
    }

    public static void onSendingMessage(String str) {
        sSendingMessages.put(str, Long.valueOf(System.currentTimeMillis()));
    }
}

File: legacy/src/main/java/com/xiaomi/push/service/profile/MessageProfiling.kt
package com.xiaomi.push.service.profile
import io.github.magisk317.mipush.protocol.model.*

import android.util.Pair
import com.xiaomi.push.mpcd.Constants
import java.util.Vector
import java.util.concurrent.ConcurrentHashMap

object MessageProfiling {
    private val sSentPerfDatas = Vector<Pair<String, Long>>()
    private val sSendingMessages = ConcurrentHashMap<String, Long>()

    @JvmStatic
    fun getPrefString(): String {
        val sb = StringBuilder()
        synchronized(sSentPerfDatas) {
            for (i in 0 until sSentPerfDatas.size) {
                val pair = sSentPerfDatas.elementAt(i)
                sb.append(pair.first)
                sb.append(":")
                sb.append(pair.second)
                if (i < sSentPerfDatas.size - 1) {
                    sb.append(Constants.ITEM_SEPARATOR)
                }
            }
            sSentPerfDatas.clear()
        }
        return sb.toString()
    }

    @JvmStatic
    fun onReceiveSentAck(messageId: String, str2: String) {
        sSendingMessages[messageId]?.let { timestamp ->
            sSentPerfDatas.add(Pair(str2, System.currentTimeMillis() - timestamp))
            sSendingMessages.remove(messageId)
        }
    }

    @JvmStatic
    fun onSendingMessage(messageId: String) {
        sSendingMessages[messageId] = System.currentTimeMillis()
    }
}
